tests/google/cloud/operators/test_kubernetes_engine.py

```diff
@@ -0,0 +1,107 @@
+from unittest import mock
+
+import pytest
+from airflow.exceptions import AirflowException, TaskDeferred
+from kubernetes.client import models as k8s
+from kubernetes.client.models.v1_object_meta import V1ObjectMeta
+
+from astronomer.providers.google.cloud.operators.kubernetes_engine import (
+    GKEStartPodOperatorAsync,
+)
+from astronomer.providers.google.cloud.triggers.kubernetes_engine import (
+    GKEStartPodTrigger,
+)
+
+PROJECT_ID = "astronomer-***-providers"
+LOCATION = "us-west1"
+GKE_CLUSTER_NAME = "provider-***-gke-cluster"
+NAMESPACE = "default"
+POD_NAME = "astro-k8s-gke-test-pod-25131a0d9cda46419099ac4aa8a4ef8f"
+GCP_CONN_ID = "google_cloud_default"
+
+
+@pytest.fixture()
+def context():
+    """
+    Creates an empty context.
+    """
+    context = {}
+    yield context
+
+
+@mock.patch(
+    "airflow.providers.google.cloud.operators.kubernetes_engine.GKEStartPodOperator.get_gke_config_file"
+)
+@mock.patch(
+    "airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator.build_pod_request_obj"
+)
+@mock.patch(
+    "airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator.get_or_create_pod"
+)
+def test__get_or_create_pod(mock_get_or_create_pod, moc_build_pod_request_obj, mock_tmp):
```

Comment:
is the double underscore intentional ?
```suggestion
def test_get_or_create_pod(mock_get_or_create_pod, moc_build_pod_request_obj, mock_tmp):
```

Reply:
yes, actually method is private to the class i.e `_get_or_create_pod`